### PR TITLE
Sessions page

### PR DESCRIPTION
**Pull request by tonywu315**

### CODE REVIEW

File: constellations/gemini-admin/src/index.js [hunk 1]
```diff
@@ -10,32 +10,44 @@ import { ProgramPage } from "./programs/program.js";
...
-const Achieve = () => <AchievePage />;
-const AchieveEdit = () => <AchieveEditPage />;
-const AchieveEditMatch = ({ match }) => (
-    <AchieveEditPage Id={match.params.Id} />
-);
-const Announce = () => <AnnouncePage />;
-const AnnounceEdit = () => <AnnounceEditPage />;
-const AnnounceEditMatch = ({ match }) => (
-    <AnnounceEditPage announceId={match.params.announceId} />
-);
```
Comment (mythicat554) [inline constellations/gemini-admin/src/index.js:31]: There isn't anything you have to change, but I'm just wondering why you chose to put the files in this particular order?

File: constellations/gemini-admin/src/session/sessionAdd.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+    render = () => {
+        return (
+            <section id="add-sessions">
+                <h3>Add Sessions</h3>
+
+                <div className="container">
+                    <div id="date-inputs">
+                        <div className="row">
+                            <div className="item">
```
Comment (mythicat554) [inline constellations/gemini-admin/src/session/sessionAdd.js:97]: Is there any way to use less divs here?

File: constellations/gemini-admin/src/index.js (hunk 1 above)
Comment (tonywu315) [reply constellations/gemini-admin/src/index.js:31]: I just put them in the same order as the tabs.

File: constellations/gemini-admin/src/session/sessionAdd.js (hunk 2 above)
Comment (tonywu315) [reply constellations/gemini-admin/src/session/sessionAdd.js:97]: Not really. The item div is necessary to keep the label and the date picker together. The row is a flexbox. The `date-inputs` holds the 4 input things together and the `container` is another flexbox that helps with formatting the button next to the inputs.

File: constellations/gemini-admin/src/index.js [hunk 3]
```diff
@@ -71,11 +77,22 @@ class App extends React.Component {
                 <Switch>
                     <Route path="/" exact component={Home} />
                     <Route
-                        path="/program/:programId/edit"
+                        path="/programs/:programId/edit"
```
Comment (ahsu1230) [inline constellations/gemini-admin/src/index.js:80]: oh nice. Thanks

File: constellations/gemini-admin/src/session/sessionEdit.sass [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        align-items: center
+
+        h4
+            margin: 0
+
+        input
+            width: $spacing2x
+            margin: 0 $spacing2x
+    
+    buttons
```
Comment (ahsu1230) [inline constellations/gemini-admin/src/session/sessionEdit.sass:40]: did you mean `button` or a class `.buttons`?

**Comment by ahsu1230** [comment]
Oooh this is coming along nicely! Here are a few things

1) When I select a class from the dropdown menu (on the sessions page), the sessions you show must be the sessions for that class. So every time I select a new classId, the session rows need to change. So you will need to call the GET sessions api on every change (use `onChange` on the <select>)

2) Similarly, when you add sessions, you are only adding sessions to the selected classId. I think you might be doing that, but it's hard to tell which sessions belong to which classId.

**Comment by ahsu1230** [comment]
Up to you if you want to do this now or in separate PR. But you can also remove some code in the classes flow. We only want to manage sessions on this page, so you can remove the sessions stuff from the classEdit page. Remove anything after the "Display Time" section.

**Comment by tonywu315** [comment]
I will create another PR for the classes page to change to sass, use flexbox, and remove the sessions part.

File: constellations/gemini-admin/src/session/session.js [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        API.post("api/sessions/create", sessionsJSON)
+            .then(() => {
+                console.log("Successfully created sessions!");
+            })
+            .catch((err) => {
+                window.alert("Could not create sessions: " + err);
+            })
+            .finally(() => {
+                this.fetchSessionData(this.state.classId);
+            });
```
Comment (ahsu1230) [inline constellations/gemini-admin/src/session/session.js:66]: very nice!

File: constellations/gemini-admin/src/session/session.js [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+                classId: classId,
+            });
+            this.fetchSessionData(this.state.classId);
+        });
+    };
+
+    fetchSessionData = (classId) => {
+        API.get("api/sessions/class/" + classId).then((res) => {
+            var newSessions = res.data;
+            newSessions = _.sortBy(newSessions, ["startsAt"]);
```
Comment (ahsu1230) [inline constellations/gemini-admin/src/session/session.js:30]: I'm surprised _.sortBy works. Usually we're supposed to import lodash functions individually. Maybe i botched the library importing... :0

File: constellations/gemini-admin/src/session/session.sass [hunk 7]
```diff
@@ -0,0 +1,19 @@
+@import "../_.constants.sass"
+
+#select-class
+    display: flex
+    flex-flow: row wrap
+    align-items: center
+    max-width: $spacing96x
+    margin: auto
+    margin-top: $spacing4x
```
Comment (ahsu1230) [inline constellations/gemini-admin/src/session/session.sass:9]: I'd just combine these into:
`margin: $spacing4x auto 0`

File: constellations/gemini-admin/src/session/session.js [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+                        id="dropdown"
+                        onChange={(e) => this.onChangeSelect(e)}>
+                        {classOptions}
+                    </select>
+                </section>
+                <SessionAdd
+                    classId={this.state.classId}
+                    addSessions={this.addSessions}
+                />
+                {sessionsList}
```
Comment (ahsu1230) [inline constellations/gemini-admin/src/session/session.js:102]: Good job splitting this into two sections. I think it's def the right call

File: constellations/gemini-admin/src/session/sessionAdd.sass [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+        margin: $spacing1x 0
+        padding: 0
+        font-size: $fontSize0
+
+    #repeat
+        width: $spacing24x
+
+    #btn-add-sessions
+        margin: $spacing2x
+        margin-right: 0
```
Comment (ahsu1230) [inline constellations/gemini-admin/src/session/sessionAdd.sass:45]: `margin: $spacing2x 0`

spacing2x for top & bottom
0 for left & right

File: constellations/gemini-admin/src/session/sessionList.sass [hunk 10]
```diff
@@ -0,0 +1,36 @@
+@import "../_.constants.sass"
+
+#session-rows
+    max-width: $spacing96x
+    margin: auto
+    margin-top: $spacing4x
```
Comment (ahsu1230) [inline constellations/gemini-admin/src/session/sessionList.sass:6]: margin: $spacing4x auto 0

File: constellations/gemini-admin/src/session/sessionList.js [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+                    <span className="large-column">{session.notes}</span>
+                    <span className="edit">
+                        <Link
+                            to={
+                                "/sessions/" +
+                                classId +
+                                "/" +
+                                session.id +
+                                "/edit"
+                            }>
```
Comment (ahsu1230) [inline constellations/gemini-admin/src/session/sessionList.js:43]: This part is a bit hard to read. Maybe save the value into a link?
```
const url = "sessions/" + classId + "/" + session.id + ...;

<Link to={url}/>
```

File: constellations/gemini-admin/src/session/sessionEdit.js [hunk 12]
```diff
@@ -0,0 +1,275 @@
...
+                            }}
+                        />
+                    </div>
+                </div>
+
+                <div id="cancel-toggle" className="item">
+                    <h4>Canceled</h4>
+                    <input
+                        type="checkbox"
+                        value={this.state.canceled}
```
Comment (ahsu1230) [inline constellations/gemini-admin/src/session/sessionEdit.js:250]: I think this is not the correct way to set a checkbox as checked or not.
You'll need something like this: https://stackoverflow.com/questions/39120007/setting-a-checkbox-check-property-in-react

You have to update the "checked" attribute of <input> in order to have it checked by default

File: constellations/gemini-admin/src/localStorage.js [hunk 13]
```diff
@@ -0,0 +1,11 @@
+const STORAGE_CURRENT_CLASS = "storage_current_class";
+
+export const getCurrentClassId = () => {
+    const classId = window.localStorage.getItem(STORAGE_CURRENT_CLASS);
+    return JSON.parse(classId);
+};
+
+export const updateCurrentClassId = (c) => {
```
Comment (ahsu1230) [inline constellations/gemini-admin/src/localStorage.js:8]: setCurrentClassId() we can mirror the localStorage API here (get & set)

File: constellations/gemini-admin/src/session/session.js [hunk 14]
```diff
@@ -0,0 +1,117 @@
...
+
+export class SessionPage extends React.Component {
+    state = {
+        classes: [],
+        classId: "",
+        sessions: [],
+    };
+
+    componentDidMount = () => {
+        console.log("WHAT");
```
Comment (ahsu1230) [inline constellations/gemini-admin/src/session/session.js:17]: remove :)

Comment (tonywu315) [reply constellations/gemini-admin/src/session/session.js:17]: oops lol

**Pull request merged**